src/components/shared/Entity/EndpointConfig/index.tsx

```diff
@@ -5,33 +5,106 @@ import EndpointConfigForm from 'components/shared/Entity/EndpointConfig/Form';
 import EndpointConfigHeader from 'components/shared/Entity/EndpointConfig/Header';
 import WrapperWithHeader from 'components/shared/Entity/WrapperWithHeader';
 import Error from 'components/shared/Error';
+import { useEntityWorkflow } from 'context/Workflow';
+import useGlobalSearchParams, {
+    GlobalSearchParams,
+} from 'hooks/searchParams/useGlobalSearchParams';
 import useConnectorTag from 'hooks/useConnectorTag';
+import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { isEqual } from 'lodash';
+import { useEffect, useMemo } from 'react';
 import { useIntl } from 'react-intl';
-import { JsonFormsData } from 'types';
+import { createJSONFormDefaults } from 'services/ajv';
+import {
+    useEndpointConfigStore_endpointConfig_data,
+    useEndpointConfigStore_setEndpointConfig,
+    useEndpointConfigStore_setEndpointSchema,
+    useEndpointConfig_setServerUpdateRequired,
+} from 'stores/EndpointConfig';
+import { Schema } from 'types';
 
 interface Props {
     connectorImage: string;
+    draftSpecs?: DraftSpecQuery[];
     readOnly?: boolean;
-    initialEndpointConfig?: JsonFormsData | null;
 }
 
 function EndpointConfig({
     connectorImage,
+    draftSpecs = [],
     readOnly = false,
-    initialEndpointConfig,
 }: Props) {
     const intl = useIntl();
 
+    const connectorId = useGlobalSearchParams(GlobalSearchParams.CONNECTOR_ID);
+
+    const workflow = useEntityWorkflow();
+    const editWorkflow =
+        workflow === 'capture_edit' || workflow === 'materialization_edit';
```

Comment:
Weird thought... should we have the checking if we're in an edit workflow _into_ the hook itself somehow?

Reply:
Not a weird thought at all. The workflow context provider was intended to be a mere _create_ or _edit_ flag. In the event a specific subset of the _create_ and _edit_ workflows need be identified, the entity type context provider should be used. As I rolled that out, there were more entity type-specific workflow checks then general workflow checks. The logic across the board has shifted considerably since that call was made, and I believe the original approach is a better fit for the logic in place today.

Reverting the workflow context provider to its origins has been a mentally logged _action item_ for a little while. Since this is a non-essential refactoring and I thought this code would land sooner, I put it off. I will correct this before merging the code in as opposed to a fast-follow.